### DISCUSSION TITLE
Warnings of missing AMD firmware

### DISCUSSION THREAD

**Discussion by manujchandra** [SpiralLinux/SpiralLinux-project#330] [Q&A] [answered]
Hi,

Firstly, thanks for a fantastic distribution.

I installed Spiral Linux 12.x (latest) on a laptop with AMD Ryzen 5 APU with NO dedicated GPU.

Whenever I run apt, I get many warnings of missing firmware.

I followed the solution given on this [page](https://askubuntu.com/questions/1124253/missing-firmware-for-amdgpu). 

`
sudo cp ~/Documents/linux-firmware/amdgpu/* /lib/firmware/amdgpu && sudo update-initramfs -k all -u -v
`

Many warnings disappeared, but some still remain, as can be seen in the screenshot.

![20231015_182040](https://github.com/SpiralLinux/SpiralLinux-project/assets/15191493/c1b6a4bd-2f5c-4838-9894-1dd5c40a8d32)

The system otherwise seems to be working OK. Please help me to learn how to remove these warnings and error. I have not made any changes to the system, its a fresh install.

Thanks!

**Comment by geckolinux** [comment]
Hi there, the real error in your screenshot is related to the `raspi-firmware` issue. It looks like you installed a SpiralLinux version that contained an upstream Debian bug, which was fixed in SpiralLinux 12.231008. You can use this solution to fix your installation:
https://forums.debian.net/viewtopic.php?p=777348#p777348

As for the `amdgpu` firmware errors, it's nothing to worry about:
https://forums.debian.net/viewtopic.php?p=724099#p724099

**Comment by manujchandra** [comment, marked as answer]
Hi @geckolinux 

I reinstalled the SpiralLinux_XFCE_12.231008_x86-64.iso, and it seems both the problems are fixed and there don't seem to be any issues.

Thanks again for a useful distro.
